python/ml4ir/applications/ranking/model/metrics/metrics_impl.py

```diff
@@ -1,14 +1,19 @@
-import tensorflow as tf
-from tensorflow.keras import metrics
-from tensorflow.python.ops import math_ops
+from typing import Optional, Dict
+
 import numpy as np
+import tensorflow as tf
 from tensorflow import Tensor
 from tensorflow import dtypes
+from tensorflow.keras import metrics
+from tensorflow.python.ops import math_ops
 
-from ml4ir.base.model.metrics.metrics_impl import MetricState
 from ml4ir.base.features.feature_config import FeatureConfig
+from ml4ir.base.model.metrics.metrics_impl import MetricState
 
-from typing import Optional, Dict
+
+class CombinationMetric:
```

Comment:
Can you remind me, why this is needed?

Reply:
This is used to distinguish single label metrics from multi-label metrics. Multi-label metrics should be computed only for one of the outputs